src/insights.ts

```diff
@@ -5,31 +5,35 @@ import type {
   InsightsRegion,
 } from './insightsAPIBeaconClient';
 import { InsightsApiBeaconClient } from './insightsAPIBeaconClient';
+import type { Expand } from './types/utils';
 import type { UserTokenOptions } from './userToken';
 import { UserToken } from './userToken';
 
 type BufferedMethodCall = [string, unknown, unknown, unknown?];
 
 type SnippetAlgoliaInsights = BufferedMethodCall[];
 
-type ObjectIDsAfterSearchEvent = {
-  eventName: string;
-  index: string;
+type BaseEvent<T> = Expand<
+  T & {
+    eventName: string;
+    index: string;
+    appId?: string;
```

Comment:
This should stay consistent with the existing options passed to the constructor.

```suggestion
    applicationId?: string;
```

Reply:
Ah I didn't notice! I'll update it in this PR and the ones for v1/v2.

Reply:
I switched to providing custom headers directly on method calls, following Ben's comment.